Swift package manifest decoding error

We have three packages that consistently fail analysis

```
[ WARNING ] App.AppError.noValidVersions(Optional(AD2D160F-E6C7-42B2-B76B-8CD71AEEFF7B), "https://github.com/maparoni/GeoJSONKit-Turf.git") [component: analyze]
[ WARNING ] App.AppError.noValidVersions(Optional(A8DA9C80-6C44-4538-A99B-35909789FF79), "https://github.com/thepotatoking55/CGLFW3.git") [component: analyze]
[ WARNING ] App.AppError.noValidVersions(Optional(7D04578E-BFF3-4C89-8676-A2FF90365B04), "https://github.com/compnerd/swift-firebase.git") [component: analyze]
```

Starting with the first one, we fail version detection, because the package manifest contains `.custom("xros", versionString: "1.0")`, which we fail to decode:

```
DecodingError.dataCorrupted: Data was corrupted. Path: platformName. Debug description: invalid rawValue: xros
```

Finally, we have

```
/Users/sas/Projects/SPI/spi-server/SPI-checkouts/github.com-compnerd-swift-firebase/Package.swift:5:8: error: no such module 'WinSDK'
  3 | import PackageDescription
  4 | import Foundation
  5 | import WinSDK
    |        `- error: no such module 'WinSDK'
  6 |
  7 | let swiftc = "swiftc.exe".withCString(encodedAs: UTF16.self) { pwszCompiler in
error: invalid manifests at [<AbsolutePath:"/Users/sas/Projects/SPI/spi-server/SPI-checkouts/github.com-compnerd-swift-firebase">]
```

Same applies here: we can't fix it.

On second thought, we may not want to suppress critical errors here just because we can't do anything about them. At least this way, we'd (more easily) see _why_ a package isn't seeing version updates.

At least in the SwiftFirebase case, it's also easy to push a fix upstream by `#ifdef` guarding the import.

--- a/Sources/App/Models/Manifest.swift
+++ b/Sources/App/Models/Manifest.swift
@@ -67,13 +67,18 @@ struct Manifest: Decodable, Equatable {
         init(from decoder: any Decoder) throws {
             let container = try decoder.container(keyedBy: CodingKeys.self)
             
-            do { // case-insensitive decoding of Platform.Name
+            do { // case-insensitive decoding and re-mapping (where required) of Platform.Name
                 let rawValue = try container.decode(String.self, forKey: CodingKeys.platformName).lowercased()
-                guard let name = Name(rawValue: rawValue) else {
-                    throw DecodingError.dataCorrupted(.init(codingPath: [CodingKeys.platformName],
-                                                            debugDescription: "invalid rawValue: \(rawValue)"))
+                switch rawValue {
+                    case "xros":
+                        self.platformName = .visionos
+                    default:
+                        guard let name = Name(rawValue: rawValue) else {
+                            throw DecodingError.dataCorrupted(.init(codingPath: [CodingKeys.platformName],
+                                                                    debugDescription: "invalid rawValue: \(rawValue)"))
+                        }
+                        self.platformName = name
                 }
-                self.platformName = name
             }
             self.version = try container.decode(String.self, forKey: CodingKeys.version)
         }

--- a/Tests/AppTests/ManifestTests.swift
+++ b/Tests/AppTests/ManifestTests.swift
@@ -21,6 +21,25 @@ import Testing
 
 extension AllTests.ManifestTests {
 
+    @Test func decode_Platform() throws {
+        struct Test: Decodable, Equatable {
+            var platform: Manifest.Platform
+        }
+        
+        let data = Data("""
+            {
+              "platform" : {
+                "options" : [
+            
+                ],
+                "platformName" : "xros",
+                "version" : "1.0"
+              }
+            }
+            """.utf8)
+        #expect(try JSONDecoder().decode(Test.self, from: data) == .init(platform: .init(platformName: .visionos, version: "1.0")))
+    }
+
     @Test func decode_Product_Type() throws {
         // Test product type decoding.
         // JSON snippets via `swift package dump-package` from the following

